heat/core/operations.py

```diff
@@ -509,6 +515,494 @@ def sum(x, axis=None, out=None):
+            rem2 = 0
+            sz = var_tot.shape[0]
+            while True:  # this loop will loop pairwise over the whole process and do pairwise updates
+                if sz % 2 != 0:
+                    if rem1 != 0 and rem2 == 0:
+                        rem2 = sz - 1
+                    elif rem1 == 0:
+                        rem1 = sz - 1
+                splt = int(sz / 2)
+                for i in range(splt):
+                    merged = merge_vars(var_tot[i, 0], var_tot[i, 1], var_tot[i, 2],
+                                        var_tot[i + splt, 0], var_tot[i + splt, 1], var_tot[i + splt, 2], bessel)
+                    for enum, m in enumerate(merged):
+                        var_tot[i, enum] = m
+                if rem1 and rem2:
+                    merged = merge_vars(var_tot[rem1, 0], var_tot[rem1, 1], var_tot[rem1, 2],
+                                        var_tot[rem2, 0], var_tot[rem2, 1], var_tot[rem2, 2], bessel)
+                    for enum, m in enumerate(merged):
+                        var_tot[rem2, enum] = m
+                    rem1 = rem2
+                    rem2 = 0
+                sz = splt
+                if sz == 1 or sz == 0:
+                    if rem1:
+                        merged = merge_vars(var_tot[0, 0], var_tot[0, 1], var_tot[0, 2],
+                                            var_tot[rem1, 0], var_tot[rem1, 1], var_tot[rem1, 2], bessel)
+                        for enum, m in enumerate(merged):
+                            var_tot[0, enum] = m
+                    return var_tot[0][0]
+        else:
+            # full matrix on one node
+            return __local_operation(torch.var, x, out=None, unbiased=bessel)
+    else:
+        # case for mean in one dimension
+        output_shape = list(x.shape)
+        if isinstance(axis, int):
+            if axis >= len(x.shape):
+                raise ValueError("axis must be < {}, currently is {}".format(len(x.shape), axis))
+            axis = axis if axis > 0 else axis % len(x.shape)
+            # only one axis given
+            output_shape = [output_shape[it] for it in range(len(output_shape)) if it != axis]
+            if x.split is None:
+                return __local_operation(torch.var, x, out=None, dim=axis, unbiased=bessel)
+            elif axis == x.split:
+                return reduce_vars_elementwise(output_shape)
+            else:
+                try:
+                    return tensor.array(__local_operation(torch.var, x, out=None, dim=axis, unbiased=bessel), split=x.split, comm=x.comm)
+                except ValueError:
+                    return __local_operation(torch.var, x, out=None, dim=axis, unbiased=bessel)
+        else:
+            raise TypeError("Axis (axis) must be an int, currently is {}. Check if multidim var is available in pyTorch".format(type(axis)))
+            # TODO: when multi dim support is available from pytorch this can be uncommented and the raise above can be changed
```

Comment:
Should this be removed or documented in another ticket?

Reply:
It could be either. the code is there when it's available in pytorch. the only changes required are to uncomment that code block and to change the raise to allow iterables.

Reply:
see issue #156